tests/update_workflow.go

```diff
@@ -1813,15 +1677,25 @@ func (s *FunctionalSuite) TestUpdateWorkflow_NewNormalWorkflowTask_Reject() {
   3 WorkflowTaskStarted
   4 WorkflowTaskCompleted
   5 ActivityTaskScheduled
-  6 WorkflowTaskScheduled // Normal WT because there is ActivityTaskScheduled(5) event.
+  6 WorkflowTaskScheduled // Speculative WFT will be written to the history because there is ActivityTaskScheduled(5) event.
   7 WorkflowTaskStarted
 `, task.History)
 			return nil, nil
 		case 3:
 			s.EqualHistory(`
-  8 WorkflowTaskCompleted
-  9 WorkflowTaskScheduled
- 10 WorkflowTaskStarted`, task.History)
+  1 WorkflowExecutionStarted
+  2 WorkflowTaskScheduled
+  3 WorkflowTaskStarted
+  4 WorkflowTaskCompleted
+  5 ActivityTaskScheduled
+  6 WorkflowTaskScheduled
+  7 WorkflowTaskStarted
+  8 WorkflowTaskCompleted // Empty speculative WFT was written to the history because it shipped events.
+  9 ActivityTaskStarted
+ 10 ActivityTaskCompleted
+ 11 WorkflowTaskScheduled
+ 12 WorkflowTaskStarted
```

Comment:
It's surprising to see that even though no implementation or test code was changed, `EqualHistory` sees two extra events and still just succeeds? I suppose I don't understand `EqualHistory` yet. I guess it's more like a `ContainsHistory`?

Reply:
No, it is actual `EqualHistory`. Test code was actually changed. Essentially it is `PollAndProcessWorkflowTask` (which returns full history if not on sticky queue) instead of `HandlePartialWorkflowTask` (which always returns short "sticky" version of history).